hooks/use-timeline.ts

```diff
@@ -0,0 +1,27 @@
+import { useInfiniteQuery } from '@tanstack/react-query';
+
+import { TimeLineResponse } from '@/features/main/time-line';
+
+const getTimeLineData = async (cursorRecordAt: unknown) => {
```

Comment:
음 정확하게는 모르겠지만, 일부러 pageParam에 대한 타입을 unknown으로 열어둔 것 같네용
https://github.com/TanStack/query/issues/6628
> this was a limitation we accepted when we implemented the type inference for pageParam in the queryFn and in getNextPageParam because we didn't want to add the complexity of more type parameters everywhere.
> 
> This is only relevant if you need the pageParams during rendering (returned from useInfiniteQuery), and I don't think that's very common. What do you need them for?